src/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -50,6 +49,16 @@ public void TryGetAddrInfo_HostName()
                 // machine configurations, which varies by distro and is often inconsistent.
                 return;
             }
+
+            // Temporary instrumentation for #32797
+            if (error == SocketError.TryAgain && Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                error = NameResolutionPal.TryGetAddrInfo(hostName, out hostEntry, out nativeErrorCode);
+                if (error != SocketError.TryAgain)
+                {
+                    throw new InvalidOperationException("Name resolution failure preventable with retry");
+                }
+            }
             
             Assert.Equal(SocketError.Success, error);
```

Comment:
The test just failed again at this assert, meaning the retry didn't help, at least not in that case:
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F34956~2Fmerge/test~2Ffunctional~2Fcli~2F/20190130.31/workItem/System.Net.NameResolution.Pal.Tests/analysis/xunit/System.Net.NameResolution.PalTests.NameResolutionPalTests~2FTryGetAddrInfo_HostName